plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/reports/class-wc-tests-reports-products.php

```diff
@@ -491,4 +696,382 @@ public function test_report_export_arguments() {
 	}
+
+	/**
+	 * Tests the data stored in the wc_order_product_lookup table when a full refund is made.
+	 *
+	 * The full refunds here are the ones that change the order status to refunded.
+	 * The refund type will be full but there will not be refund order line items.
+	 */
+	public function test_sync_order_products_full_refund() {
+		global $wpdb;
+
+		WC_Helper_Reports::reset_stats_dbs();
+
+		// Enable Tax.
+		update_option( 'woocommerce_prices_include_tax', 'no' );
+		update_option( 'woocommerce_calc_taxes', 'yes' );
+
+		// Create a 10% tax rate.
+		$tax_rate = array(
+			'tax_rate_country'  => '',
+			'tax_rate'          => '10',
+			'tax_rate_name'     => 'tax',
+			'tax_rate_order'    => '0',
+			'tax_rate_shipping' => '1',
+		);
+		WC_Tax::_insert_tax_rate( $tax_rate );
+
+		// Create product 1 with price $25.
+		$product_1 = new WC_Product_Simple();
+		$product_1->set_name( 'Test Product 1' );
+		$product_1->set_regular_price( 25 );
+		$product_1->save();
+
+		// Create product 2 with price $30.
+		$product_2 = new WC_Product_Simple();
+		$product_2->set_name( 'Test Product 2' );
+		$product_2->set_regular_price( 30 );
+		$product_2->save();
+
+		// Create an order and add product_1 as the order item. The quantity is set to 4.
+		$order = WC_Helper_Order::create_order( 1, $product_1 );
+
+		// Add product_2 as the second order item. The quantity is set to 2.
+		$item = new WC_Order_Item_Product();
+		$item->set_props(
+			array(
+				'product'  => $product_2,
+				'quantity' => 2,
+				'subtotal' => wc_get_price_excluding_tax( $product_2, array( 'qty' => 2 ) ),
+				'total'    => wc_get_price_excluding_tax( $product_2, array( 'qty' => 2 ) ),
+			)
+		);
+		$item->save();
+		$order->add_item( $item );
+
+		// Add a flat rate shipping method to the order. The shipping cost is $100.
+		$rate          = new WC_Shipping_Rate( 'flat_rate_shipping', 'Flat rate shipping', '100', array(), 'flat_rate' );
+		$shipping_item = new WC_Order_Item_Shipping();
+		$shipping_item->set_props(
+			array(
+				'method_title' => $rate->label,
+				'method_id'    => $rate->id,
+				'total'        => wc_format_decimal( $rate->cost ),
+				'taxes'        => $rate->taxes,
+			)
+		);
+		foreach ( $rate->get_meta_data() as $key => $value ) {
+			$shipping_item->add_meta_data( $key, $value, true );
+		}
+		// Remove existing shipping items that created by WC_Helper_Order::create_order.
+		$order->remove_order_items( 'shipping' );
+		$order->add_item( $shipping_item );
+
+		$order->set_status( OrderStatus::COMPLETED );
+		$order->calculate_totals( true );
+		$order->save();
+
+		WC_Helper_Queue::run_all_pending( 'wc-admin-data' );
+
+		// Refund the order completely by changing the order status to refunded.
+		$order->set_status( OrderStatus::REFUNDED );
+		$order->save();
+
+		WC_Helper_Queue::run_all_pending( 'wc-admin-data' );
+
+		// Get the last order item from the order.
+		$order_item_id = end( $order->get_items() )->get_id();
```

Comment:
I get an error when trying to run this test locally:

> Only variables should be passed by reference

```suggestion
		$order_items   = $order->get_items();
		$order_item_id = end( $order_items )->get_id();
```

I understand that in GitHub actions it's running with `wp-env` which is different then I run it. Although it's not clear to me why the same error wouldn't be reported there. I tried testing with different PHP versions and I get the same error all the way from 7.4 - 8.3

There are a few locations in this file where we call `end` on the response of a method call.